### PR TITLE
Update a few things

### PR DESCRIPTION
**Pull request by Stanzilla**
Finally got around to rebasing all my previous work into a hopefully stable place. This time I haven't touched your code at all, so you can still decide if you want to apply the new linting rules or not. The CI builds will fail because of that.

So this PR does:

* Update eslint and set most of the new rules it comes with to `0` aka ignore.
* Changes the travis and appveyor scripts to use yarn for everything
* Changes the travis and appveyor scripts to use node 7
* Updates electron and companion dependencies

### CODE REVIEW

**Comment by KELiON** [comment]
@Stanzilla just a quick question – can we update everything, but not eslint?

**Comment by Stanzilla** [comment]
@KELiON sure! Like this? I had a bit of trouble trying to figure out why the Travis builds were failing, but it was because it tried to build 32bit builds on a 64bit machine. I tried to make it work but noticed that 32bit Linux builds were not even part of your Cerebro releases so I just removed the 32bit arch, is that ok? Guess it never failed before because the format for electron-builder was wrong and it fell back to default, which is 64bit only.

**Comment by KELiON** [comment]
@Stanzilla actually, some people were asking about 32 bit builds! Can you add it in separate PR? It would be great!

File: app/main/plugins/core/osx-apps/lib/mdfind.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ export default function mdfind({
       .pipe(map(filterEmpty))
       // eslint-disable-next-line func-names
       .pipe(through(function (data) {
-        times++
+        times += times
```
Comment (KELiON) [inline app/main/plugins/core/osx-apps/lib/mdfind.js]: It should be `time += 1`, `times += times` doubles the value

**Comment by KELiON** [comment]
I fixed it by myself. THanks for PR, @Stanzilla!

**Comment by Stanzilla** [comment]
Yay! And my Linux knowledge is 0, I tried a few things but could not get it to work, sorry :(

**Pull request merged**